### PR TITLE
Add bounds check on allocation size.

### LINKED ISSUE #853: Out of memory error due to unchecked allocation size in JpegBase::readMetadata()
There is an out-of-memory error at [jpgimage.cpp:484](https://github.com/Exiv2/exiv2/blob/c0ecc2ae36f34462be98623deb85ba1747ae2175/src/jpgimage.cpp#L484) because there is no bounds check on the value of `s`.

issue_853_poc

Reproducible on the 0.27-maintenance branch ([c0ecc2a](https://github.com/Exiv2/exiv2/commit/c0ecc2ae36f34462be98623deb85ba1747ae2175)):

```
exiv2 issue_853_poc.jpg
```

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Fixes #853.

### CODE REVIEW

File: src/jpgimage.cpp [hunk 1]
```diff
@@ -459,6 +460,10 @@ namespace Exiv2 {
                 --search;
             }
             else if ( marker == app2_ && memcmp(buf.pData_ + 2, iccId_,11)==0) {
+                if (size < 2+14) {
+                    rc = 3;
```
Comment (kevinbackhouse) [inline src/jpgimage.cpp]: I am not sure if the value of `rc` is important. I copied this value (`rc = 3`) from a few lines up.

Comment (D4N) [reply src/jpgimage.cpp]: The `rc` value is only pretends to be a return value, it is just used in line 558 to print a warning. I'd suggest to assign a value larger than 7 to `rc`, as that is not yet used.

Comment (kevinbackhouse) [reply src/jpgimage.cpp]: Thanks. I have changed it to 8.

**Review by D4N — DISMISSED** [review]

**Comment by D4N** [comment]
@piponazo can you restart the Appveyor build?

**Comment by piponazo** [comment]
> @piponazo can you restart the Appveyor build?

Sure. What I am doing with all @kevinbackhouse 's branches is to click in "Update branch" (to merge 0.27-maintenance into his branch) and then I squash the commits before merging. In that way, we do not need the the contributor's intervention for every PR and we do not pollute the git history either.

**Review by piponazo — APPROVED** [review]

**Pull request merged**